src/Interpreters/executeQuery.cpp

```diff
@@ -291,30 +291,11 @@ static void onExceptionBeforeStart(const String & query_for_logging, ContextPtr
         if (auto query_log = context->getQueryLog())
             query_log->add(elem);
 
-    if (auto opentelemetry_span_log = context->getOpenTelemetrySpanLog();
-        context->query_trace_context.trace_id != UUID()
-            && opentelemetry_span_log)
-    {
-        OpenTelemetrySpanLogElement span;
-        span.trace_id = context->query_trace_context.trace_id;
-        span.span_id = context->query_trace_context.span_id;
-        span.parent_span_id = context->getClientInfo().client_trace_context.span_id;
-        span.operation_name = "query";
-        span.start_time_us = current_time_us;
-        span.finish_time_us = time_in_microseconds(std::chrono::system_clock::now());
-        span.attributes.reserve(6);
-        span.attributes.push_back(Tuple{"clickhouse.query_status", "ExceptionBeforeStart"});
-        span.attributes.push_back(Tuple{"db.statement", elem.query});
-        span.attributes.push_back(Tuple{"clickhouse.query_id", elem.client_info.current_query_id});
-        span.attributes.push_back(Tuple{"clickhouse.exception", elem.exception});
-        span.attributes.push_back(Tuple{"clickhouse.exception_code", toString(elem.exception_code)});
-        if (!context->query_trace_context.tracestate.empty())
-        {
-            span.attributes.push_back(Tuple{"clickhouse.tracestate", context->query_trace_context.tracestate});
-        }
-
-        opentelemetry_span_log->add(span);
-    }
+    query_span->addAttribute("clickhouse.exception_code", elem.exception_code);
+    query_span->addAttribute("clickhouse.exception", elem.exception);
+    query_span->addAttribute("db.statement", elem.query);
```

Comment:
Just wondering if "db.statement" is standardized by OpenTel or why we don't use key "clickhouse.statement" here?

Reply:
db.statement is defined in OpenTel, see: https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/database/

Reply:
Good to know, thanks.